diff --git a/packages/gatsby/cache-dir/static-query.js b/packages/gatsby/cache-dir/static-query.js
index e31cebb08a..cef8ab70ed 100644
--- a/packages/gatsby/cache-dir/static-query.js
+++ b/packages/gatsby/cache-dir/static-query.js
@@ -79,4 +79,4 @@ useStaticQuery(graphql\`${query}\`);
   }
 }
 
-export { StaticQuery, StaticQueryContext, useStaticQuery }
+export { StaticQueryContext, useStaticQuery }
