diff --git a/packages/gatsby/cache-dir/gatsby-browser-entry.js b/packages/gatsby/cache-dir/gatsby-browser-entry.js
index 0b64e8d08c..5eed580bf3 100644
--- a/packages/gatsby/cache-dir/gatsby-browser-entry.js
+++ b/packages/gatsby/cache-dir/gatsby-browser-entry.js
@@ -23,7 +23,6 @@ export {
 
 export { graphql, prefetchPathname }
 export {
-  StaticQuery,
   StaticQueryContext,
   useStaticQuery,
   StaticQueryServerContext,
