diff --git a/packages/gatsby/cache-dir/gatsby-browser-entry.js b/packages/gatsby/cache-dir/gatsby-browser-entry.js
index c2bfd3e2b7..9e0f1205bf 100644
--- a/packages/gatsby/cache-dir/gatsby-browser-entry.js
+++ b/packages/gatsby/cache-dir/gatsby-browser-entry.js
@@ -22,5 +22,5 @@ export {
 } from "gatsby-link"
 
 export { graphql, prefetchPathname }
-export { StaticQuery, StaticQueryContext, useStaticQuery } from "./static-query"
+export { StaticQueryContext, useStaticQuery } from "./static-query"
 export * from "gatsby-script"
diff --git a/packages/gatsby/cache-dir/static-query.js b/packages/gatsby/cache-dir/static-query.js
index e31cebb08a..cef8ab70ed 100644
--- a/packages/gatsby/cache-dir/static-query.js
+++ b/packages/gatsby/cache-dir/static-query.js
@@ -79,4 +79,4 @@ useStaticQuery(graphql\`${query}\`);
   }
 }
 
-export { StaticQuery, StaticQueryContext, useStaticQuery }
+export { StaticQueryContext, useStaticQuery }
