diff --git a/packages/gatsby-link/package.json b/packages/gatsby-link/package.json
index 27eec7cf4d..8e4a5f6c04 100644
--- a/packages/gatsby-link/package.json
+++ b/packages/gatsby-link/package.json
@@ -39,7 +39,7 @@
     "prepend-directive": "^1.0.3"
   },
   "peerDependencies": {
-    "@gatsbyjs/reach-router": "^1.3.5",
+    "@gatsbyjs/reach-router": "^2.0.0-v2.0",
     "react": "^18.0.0 || ^0.0.0",
     "react-dom": "^18.0.0 || ^0.0.0"
   },
diff --git a/packages/gatsby-react-router-scroll/package.json b/packages/gatsby-react-router-scroll/package.json
index 3c1eab83ac..72cdfe4955 100644
--- a/packages/gatsby-react-router-scroll/package.json
+++ b/packages/gatsby-react-router-scroll/package.json
@@ -25,7 +25,7 @@
   "license": "MIT",
   "main": "index.js",
   "peerDependencies": {
-    "@gatsbyjs/reach-router": "^1.3.5",
+    "@gatsbyjs/reach-router": "^2.0.0-v2.0",
     "react": "^18.0.0 || ^0.0.0",
     "react-dom": "^18.0.0 || ^0.0.0"
   },
diff --git a/packages/gatsby-script/package.json b/packages/gatsby-script/package.json
index 530ea7f4a6..a159a2de37 100644
--- a/packages/gatsby-script/package.json
+++ b/packages/gatsby-script/package.json
@@ -22,7 +22,7 @@
     "clean": "del-cli dist/*"
   },
   "devDependencies": {
-    "@gatsbyjs/reach-router": "^1.3.9",
+    "@gatsbyjs/reach-router": "^2.0.0-v2.0",
     "@testing-library/react": "^11.2.7",
     "cross-env": "^7.0.3",
     "del-cli": "^5.0.0",
@@ -31,7 +31,7 @@
     "typescript": "^4.7.4"
   },
   "peerDependencies": {
-    "@gatsbyjs/reach-router": "^1.3.5",
+    "@gatsbyjs/reach-router": "^2.0.0-v2.0",
     "react": "^18.0.0 || ^0.0.0",
     "react-dom": "^18.0.0 || ^0.0.0"
   },
diff --git a/packages/gatsby/cache-dir/__tests__/find-path.js b/packages/gatsby/cache-dir/__tests__/find-path.js
index da798c7f95..e47e74d600 100644
--- a/packages/gatsby/cache-dir/__tests__/find-path.js
+++ b/packages/gatsby/cache-dir/__tests__/find-path.js
@@ -93,10 +93,10 @@ describe(`find-path`, () => {
 
     it(`should only process a request once`, () => {
       jest.resetModules()
-      jest.mock(`@gatsbyjs/reach-router/lib/utils`)
+      jest.mock(`@gatsbyjs/reach-router`)
       const findPath = require(`../find-path`).findPath
       const setMatchPaths = require(`../find-path`).setMatchPaths
-      const pick = require(`@gatsbyjs/reach-router/lib/utils`).pick
+      const pick = require(`@gatsbyjs/reach-router`).pick
 
       setMatchPaths([
         {
diff --git a/packages/gatsby/cache-dir/find-path.js b/packages/gatsby/cache-dir/find-path.js
index a231735ac6..a51e05fe56 100644
--- a/packages/gatsby/cache-dir/find-path.js
+++ b/packages/gatsby/cache-dir/find-path.js
@@ -1,4 +1,4 @@
-import { pick } from "@gatsbyjs/reach-router/lib/utils"
+import { pick } from "@gatsbyjs/reach-router"
 import stripPrefix from "./strip-prefix"
 import normalizePagePath from "./normalize-page-path"
 import { maybeGetBrowserRedirect } from "./redirect-utils.js"
diff --git a/packages/gatsby/package.json b/packages/gatsby/package.json
index 188d171687..baece368fd 100644
--- a/packages/gatsby/package.json
+++ b/packages/gatsby/package.json
@@ -19,7 +19,7 @@
     "@babel/traverse": "^7.15.4",
     "@babel/types": "^7.15.4",
     "@builder.io/partytown": "^0.5.2",
-    "@gatsbyjs/reach-router": "^1.3.9",
+    "@gatsbyjs/reach-router": "^2.0.0-v2.0",
     "@gatsbyjs/webpack-hot-middleware": "^2.25.2",
     "@graphql-codegen/add": "^3.1.1",
     "@graphql-codegen/core": "^2.5.1",
diff --git a/packages/gatsby/src/bootstrap/__tests__/requires-writer.js b/packages/gatsby/src/bootstrap/__tests__/requires-writer.js
index b85af8c2f6..18c375a7b7 100644
--- a/packages/gatsby/src/bootstrap/__tests__/requires-writer.js
+++ b/packages/gatsby/src/bootstrap/__tests__/requires-writer.js
@@ -1,6 +1,6 @@
 const { joinPath } = require(`gatsby-core-utils`)
 const requiresWriter = require(`../requires-writer`)
-const { match } = require(`@gatsbyjs/reach-router/lib/utils`)
+const { match } = require(`@gatsbyjs/reach-router`)
 
 const now = Date.now()
 
diff --git a/packages/gatsby/src/bootstrap/requires-writer.ts b/packages/gatsby/src/bootstrap/requires-writer.ts
index 2c2ea7a3ea..d3457e0bff 100644
--- a/packages/gatsby/src/bootstrap/requires-writer.ts
+++ b/packages/gatsby/src/bootstrap/requires-writer.ts
@@ -4,7 +4,7 @@ import fs from "fs-extra"
 import crypto from "crypto"
 import { slash } from "gatsby-core-utils"
 import reporter from "gatsby-cli/lib/reporter"
-import { match } from "@gatsbyjs/reach-router/lib/utils"
+import { match } from "@gatsbyjs/reach-router"
 import { joinPath } from "gatsby-core-utils"
 import { store, emitter } from "../redux/"
 import { IGatsbyState, IGatsbyPage } from "../redux/types"
diff --git a/packages/gatsby/src/commands/serve.ts b/packages/gatsby/src/commands/serve.ts
index 03d8a87321..b457cde7e7 100644
--- a/packages/gatsby/src/commands/serve.ts
+++ b/packages/gatsby/src/commands/serve.ts
@@ -4,7 +4,7 @@ import fs from "fs-extra"
 import compression from "compression"
 import express from "express"
 import chalk from "chalk"
-import { match as reachMatch } from "@gatsbyjs/reach-router/lib/utils"
+import { match as reachMatch } from "@gatsbyjs/reach-router"
 import onExit from "signal-exit"
 import report from "gatsby-cli/lib/reporter"
 import telemetry from "gatsby-telemetry"
diff --git a/packages/gatsby/src/internal-plugins/functions/middleware.ts b/packages/gatsby/src/internal-plugins/functions/middleware.ts
index 23b6c6709a..a17ce67742 100644
--- a/packages/gatsby/src/internal-plugins/functions/middleware.ts
+++ b/packages/gatsby/src/internal-plugins/functions/middleware.ts
@@ -1,4 +1,4 @@
-import { match as reachMatch } from "@gatsbyjs/reach-router/lib/utils"
+import { match as reachMatch } from "@gatsbyjs/reach-router"
 import cookie from "cookie"
 import { urlencoded, text, json, raw } from "express"
 import type { RequestHandler, Request, Response, NextFunction } from "express"
diff --git a/packages/gatsby/src/utils/find-page-by-path.ts b/packages/gatsby/src/utils/find-page-by-path.ts
index 9cf7f80e21..76a248d1c6 100644
--- a/packages/gatsby/src/utils/find-page-by-path.ts
+++ b/packages/gatsby/src/utils/find-page-by-path.ts
@@ -1,5 +1,5 @@
 import { IGatsbyPage, IGatsbyState } from "../redux/types"
-import { pick } from "@gatsbyjs/reach-router/lib/utils"
+import { pick } from "@gatsbyjs/reach-router"
 
 // Ranks and picks the best page to match. Each segment gets the highest
 // amount of points, then the type of segment gets an additional amount of
diff --git a/packages/gatsby/src/utils/get-server-data.ts b/packages/gatsby/src/utils/get-server-data.ts
index 82eab8fef4..730549be4c 100644
--- a/packages/gatsby/src/utils/get-server-data.ts
+++ b/packages/gatsby/src/utils/get-server-data.ts
@@ -1,6 +1,6 @@
 import type { Request } from "express"
 import type { IGatsbyPage } from "../redux/types"
-import { match } from "@gatsbyjs/reach-router/lib/utils"
+import { match } from "@gatsbyjs/reach-router"
 
 export interface IServerData {
   headers?: Record<string, string>
