diff --git a/packages/gatsby/index.d.ts b/packages/gatsby/index.d.ts
index 57c92929c1..dae8617ee5 100644
--- a/packages/gatsby/index.d.ts
+++ b/packages/gatsby/index.d.ts
@@ -1227,6 +1227,14 @@ export interface Actions {
     option?: ActionOptions
   ): void
 
+  /** @see https://www.gatsbyjs.com/docs/reference/config-files/actions/#createSlice */
+  createSlice<TContext = Record<string, unknown>>(
+    this: void,
+    args: SliceInput<TContext>,
+    plugin?: ActionPlugin,
+    option?: ActionOptions
+  ): void
+
   /** @see https://www.gatsbyjs.com/docs/actions/#deleteNode */
   deleteNode(node: NodeInput, plugin?: ActionPlugin): void
 
@@ -1664,6 +1672,12 @@ export interface Page<TContext = Record<string, unknown>> {
   defer?: boolean
 }
 
+export interface SliceInput<TContext = Record<string, unknown>> {
+  id: string
+  component: string
+  context?: TContext
+}
+
 export interface IPluginRefObject {
   resolve: string
   options?: IPluginRefOptions
